app/views/posts/show.html.erb

```diff
@@ -23,11 +23,15 @@
 
 <div class="user-comments-container">
   <ul>
-    <% @post.comments.each do |comment| %>
-      <li>
-        <strong><%= comment.user.name %>:</strong> <%= comment.text %>
-      </li>
-    <% end %>
+  <% @post.comments.each do |comment| %>
+    <li>
+      <strong><%= comment.user.name %>:</strong> <%= comment.text %>
+      <% if can? :destroy, comment %>
+      <%= button_to 'Delete Comment', user_post_comment_path(comment.user, comment.post.id, comment), method: :delete, data: { confirm: "Are you sure?" }, class: 'btn btn-danger' %>
+      <% end %>
```

Comment:
- [ ] Kindly look into your app so that it works as required. From the project guidelines, only the admin and the user who has the comment should be able to see the delete button, but all users can see this button in your app. Kindly resolve this. Ensure that only the admin and the user who has the comment can see the button, and only these two can delete the comment.